libctru/include/3ds/services/y2r.h

```diff
@@ -1,113 +1,361 @@
+/**
+ * @file y2r.h
+ */
 #pragma once
 #include <3ds/types.h>
 
+/**
+ * @brief Input color formats
+ *
+ * For the 16-bit per component formats, bits 31-16 are padding and 15-0 contains the value.
+ */
 typedef enum
 {
-	INPUT_YUV422_INDIV_8 = 0x0,
-	INPUT_YUV420_INDIV_8 = 0x1,
-	INPUT_YUV422_INDIV_16 = 0x2,
-	INPUT_YUV420_INDIV_16 = 0x3,
-	INPUT_YUV422_BATCH = 0x4,
+    INPUT_YUV422_INDIV_8  = 0x0, ///<  8-bit per component, planar YUV 4:2:2, 32bpp, (1 Cr & Cb sample per 2x1 Y samples).\n Usually named YUV422P.
+    INPUT_YUV420_INDIV_8  = 0x1, ///<  8-bit per component, planar YUV 4:2:0, 12bpp, (1 Cr & Cb sample per 2x2 Y samples).\n Usually named YUV420P.
+    INPUT_YUV422_INDIV_16 = 0x2, ///< 16-bit per component, planar YUV 4:2:2, 16bpp, (1 Cr & Cb sample per 2x1 Y samples).\n Usually named YUV422P16.
+    INPUT_YUV420_INDIV_16 = 0x3, ///< 16-bit per component, planar YUV 4:2:0, 24bpp, (1 Cr & Cb sample per 2x2 Y samples).\n Usually named YUV420P16.
+    INPUT_YUV422_BATCH    = 0x4, ///<  8-bit per component, packed YUV 4:2:2, 16bpp, (Y0 Cb Y1 Cr).\n Usually named YUYV422.
```

Comment:
This one takes exactly the same amount of storage as YUV422P, your bpp math is wrong somewhere.